src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -3713,6 +3713,22 @@ theorem toNat_twoPow (w : Nat) (i : Nat) : (twoPow w i).toNat = 2^i % 2^w := by
     have h1 : 1 < 2 ^ (w + 1) := Nat.one_lt_two_pow (by omega)
     rw [Nat.mod_eq_of_lt h1, Nat.shiftLeft_eq, Nat.one_mul]
 
+theorem toNat_twoPow_of_le {i w : Nat} (h : w ≤ i): (twoPow w i).toNat = 0 := by
+  rw [toNat_twoPow]
+  apply Nat.mod_eq_zero_of_dvd
+  exact Nat.pow_dvd_pow_iff_le_right'.mpr h
+
+theorem toNat_twoPow_of_lt {i w : Nat} (h : i < w ): (twoPow w i).toNat = 2^i := by
+  rw [toNat_twoPow]
+  apply Nat.mod_eq_of_lt
+  apply Nat.pow_lt_pow_of_lt (by omega) (by omega)
+
+theorem toNat_twoPow_eq_if {i w : Nat} : (twoPow w i).toNat = if i < w then 2^i else 0 := by
```

Comment:
I would like for us to contemplate, in a subsequent PR, whether this should be the `simp` normal form of `toNat_twoPow`.

Reply:
I looked into this, but this breaks a couple of proofs in non-trivial ways, as several proofs use `% 2 ^ ?` instead of an if statement. Happy to investigate this, but given that this is not a clear-cut, we should investigate this more broadly.

Reply:
Happy to look into this in a subsequent PR, I'd get this PR moving in the meantime.